pose_evaluation/metrics/embedding_distance_metric.py

```diff
@@ -0,0 +1,104 @@
+from typing import Literal, Union, List
+import torch
+import torch.nn.functional as F
+import numpy as np
+
+from pose_evaluation.metrics.base_embedding_metric import EmbeddingMetric
+
+
+class EmbeddingDistanceMetric(EmbeddingMetric):
+    def __init__(self, kind: Literal["cosine", "l2"] = "cosine", device: Union[torch.device, str] = None):
+        """
+        Initialize the embedding distance metric.
+
+        Args:
+            kind (Literal["cosine", "l2"]): The type of distance metric.
+            device (torch.device | str): The device to use for computation. If None, automatically detects.
+        """
+        super().__init__(f"EmbeddingDistanceMetric {kind}", higher_is_better=False)
+        self.kind = kind
+        if device is None:
+            self.device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        else:
+            self.device = torch.device(device) if isinstance(device, str) else device
+
+    def _to_tensor(self, data: Union[np.ndarray, torch.Tensor]) -> torch.Tensor:
+        """
+        Convert input to a PyTorch tensor if it is a NumPy array.
+
+        Args:
+            data (np.ndarray | torch.Tensor): Input data.
+
+        Returns:
+            torch.Tensor: Tensor on the correct device.
+        """
+        if isinstance(data, np.ndarray):
+            data = torch.tensor(data, dtype=torch.float32)
+        return data.to(self.device)
+
+    def score(self, hypothesis: Union[np.ndarray, torch.Tensor], reference: Union[np.ndarray, torch.Tensor]) -> float:
```

Comment:
now that you have `score_all`, you can use it instead of reimplementing
```py
def score(...):
  return self.score_all([hyp], [ref])
```
you might want to access some index etc

Reply:
Oh, great idea!